client/my-sites/earn/ads/form-settings.jsx

```diff
@@ -17,76 +16,103 @@ import FormTextInput from 'calypso/components/forms/form-text-input';
+	const [ settings, setSettings ] = useState( {} );
+	const [ isChanged, setIsChanged ] = useState( false );
+
+	const site = useSelector( ( state ) => getSelectedSite( state ) );
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const siteUrl = useSelector( ( state ) => getSiteUrl( state, siteId ) );
+	const siteIsJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+	const isSavingSettings = useSelector( ( state ) => isSavingWordadsSettings( state, siteId ) );
+	const wordadsSettings = useSelector( ( state ) => getWordadsSettings( state, siteId ) );
+	const widgetsUrl = useSelector( ( state ) => getCustomizerUrl( state, siteId, 'widgets' ) );
+
+	const isLoading = ! wordadsSettings || isSavingSettings;
+	const isWordAds = site?.options?.wordads;
+
+	useEffect( () => {
+		// Set default settings if settings are empty
+		const settingsAreEmptyAndWordadsSettingsExist =
+			isEmpty( settings ) && ! isEmpty( wordadsSettings );
+
+		if ( settingsAreEmptyAndWordadsSettingsExist ) {
+			setSettings( {
+				...defaultSettings(),
 				...wordadsSettings,
 			} );
 		}
-	}
+	}, [ settings, wordadsSettings ] );
+
+	useEffect( () => {
+		// siteId has changed, so we reset settings
+		// This code is here to directly mimic code that
+		// was in UNSAFE_componentWillReceiveProps
+		setSettings( {
+			...defaultSettings(),
+			...wordadsSettings,
+		} );
+
+		// Only run on siteId change
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ siteId ] );
 
-	handleChange = ( event ) => {
+	function handleChange( event ) {
 		const name = event.currentTarget.name;
 		const value = event.currentTarget.value;
 
-		this.setState( {
+		setSettings( ( prevState ) => ( {
+			...prevState,
 			[ name ]: value,
-		} );
-	};
+		} ) );
+	}
 
-	handleToggle = ( event ) => {
+	function handleToggle( event ) {
 		const name = event.currentTarget.name;
 
-		this.setState( {
-			[ name ]: ! this.state[ name ],
-		} );
-	};
+		setSettings( ( prevState ) => ( {
+			...prevState,
+			[ name ]: ! settings[ name ],
+		} ) );
+	}
 
-	handleDisplayToggle = ( name ) => () => {
-		this.setState( ( prevState ) => ( {
+	function handleDisplayToggle( name ) {
+		setSettings( ( prevState ) => ( {
+			...prevState,
 			display_options: {
 				...prevState.display_options,
-				[ name ]: ! this.state.display_options[ name ],
+				[ name ]: ! settings.display_options[ name ],
 			},
 		} ) );
-	};
+	}
 
-	handleCompactToggle = ( name ) => () => {
-		this.setState( {
-			[ name ]: ! this.state[ name ],
-		} );
-	};
+	function handleCompactToggle( name ) {
+		setSettings( ( prevState ) => ( {
+			...prevState,
+			[ name ]: ! settings[ name ],
+		} ) );
+	}
 
-	handleSubmit = ( event ) => {
-		const { site } = this.props;
+	function handleSubmit( event ) {
 		event.preventDefault();
 
-		this.props.saveWordadsSettings( site.ID, this.packageState() );
+		dispatch( saveWordadsSettings( site.ID, packageState() ) );
```

Comment:
dispatch() needed here because in the original code the saveWordadsSettings was passed through matchDispatchtoProps.